utils/awsFunctions.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    IdentitystoreClient,
+    ListUsersCommand,
+    CreateUserCommand,
+    CreateGroupMembershipCommand,
+    ListUsersCommandInput,
+    CreateUserCommandInput,
+    CreateGroupMembershipCommandInput,
+  } from "@aws-sdk/client-identitystore";
+  import config from "config";
+  
+  // Define the configuration variables with proper types
+  const accessKeyId: string = config.get<string>("aws.access_key");
+  const secretAccessKey: string = config.get<string>("aws.secret_key");
+  const region: string = config.get<string>("aws.region");
+  const identityStoreId: string = config.get<string>("aws.identity_store_id");
+  
+  let client: IdentitystoreClient;
+  
+  /**
+   * Configures AWS SDK credentials and returns a singleton IdentitystoreClient instance.
+   * @returns {IdentitystoreClient} Singleton AWS Identitystore client
+   */
+  function configureAWSCredentials(): IdentitystoreClient {
+    if (!client) {
+      if (accessKeyId && secretAccessKey) {
+        client = new IdentitystoreClient({
+          region: region,
+          credentials: {
+            accessKeyId: accessKeyId,
+            secretAccessKey: secretAccessKey,
+          },
+        });
+      }
+    }
+    return client!;
+  }
+  
+  /**
+   * Lists users from AWS Identity Store filtered by username.
+   * @param {string} username - Username to filter users by
+   * @returns {Promise<string | null>} - UserId if found, otherwise null
+   */
+  export const fetchUserIdFromUsername = async (username: string): Promise<string | null> => {
```

Comment:
How is this method different from `fetchAwsUserIdByUsername` method?

Reply:
will remove this, initially thought to have this logic a bit different, but now its not needed

Reply:
removed this